server/metrics/relay.go

```diff
@@ -0,0 +1,132 @@
+package metrics
+
+import (
+	"context"
+	"github.com/jitsucom/jitsu/server/resources"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/carlmjohnson/requests"
+	"github.com/jitsucom/jitsu/server/logging"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+)
+
+var HashedRelayLabels = map[string]bool{
+	"source_id":      true,
+	"destination_id": true,
+}
+
+type RelayTrigger interface {
+	Channel() <-chan time.Time
+	Now() time.Time
+	Stop()
+}
+
+type TickerTrigger struct {
+	*time.Ticker
+}
+
+func (t TickerTrigger) Now() time.Time {
+	return time.Now()
+}
+
+func (t TickerTrigger) Channel() <-chan time.Time {
+	return t.C
+}
+
+type Relay struct {
+	URL          string
+	HostID       string
+	DeploymentID string
+	Timeout      time.Duration
+	work         sync.WaitGroup
+	cancel       func()
+}
+
+func (r *Relay) Run(ctx context.Context, trigger RelayTrigger, gatherer prometheus.Gatherer) {
+	r.Stop()
+	ctx, cancel := context.WithCancel(ctx)
+	r.cancel = cancel
+	r.work.Add(1)
+	go func() {
```

Comment:
Let's use `safego` package for running all goroutines into the safe wrapper which has already had panic handler.

Reply:
Fixed.